File: hw1/java/LinkedListDriver.java
public class LinkedListDriver{
	
	public static void main(String[] args){
		
		//LinkedList list = new LinkedList();
		LinkedList list = new LinkedList();
		//LinkedList another = new LinkedList();
		/*
		Node top = new Node();
		top.value = 1;
		top.next = new Node();
		top.next.value = 2;
		top.next.next = new Node();
		top.next.next.value = 3;
		top.next.next.next = null;
		*/
		
		Node top = list.initNode(0);
		list.append(top);
		list.appendData(1);
		list.appendData(2);
		list.appendData(3);
		Node another = list.initNode(9999);
		
		System.out.println("report: " + list.report());
		System.out.println("size: " + list.size());
		System.out.println("contains 1?: " + list.contains(1));
		
		list.remove(3);
		System.out.println("remove index 3: " + list.report());

		list.insert(3, another);
		list.insertData(1, 765432);
		System.out.println("\nadding 9999: " + list.report());
		
		
	}
}

File: hw1/linked_list_driver.cpp
#include <iostream>
#include "linked_list.h"

using namespace std;

int main () {



  cout << "Testing Linked List" << endl;
  node* empty_list = NULL; // empty list
  cout << "Empty List Contents: " << report(empty_list) << endl; // should be ''
  append_data(&empty_list, 16);
  cout << "Empty List Contents after appending 16: " 
       << report(empty_list) << endl; // should be '16'
       
  cout << "Size (should be 1): " << size(empty_list) << endl;


  // now lets make a proper list that we manipulate
  node* head = init_node(0);
  // append a number
  append_data(&head, 1);
  // append a few more
  
  append_data(&head, 2);
  append_data(&head, 3);
  // Now try appending nodes
  node* seven = init_node(4);
  node* ten = init_node(5);
  append(&head, seven);
  append(&head, ten);
 
  cout << "size of the list: " << size(head) << endl;	
	
	
	//node* fun = init_node(99999);
	insert_data(&head, 6, 99999);
	cout << "report (should say 0 1 2 3 4 5 99999): " << report(head) << endl;
	
	

}
